nominal/sdk.py

```diff
@@ -210,24 +204,54 @@ def update(
             name=name,
             properties=None if properties is None else dict(properties),
         )
-        response = self._catalog_client.update_dataset_metadata(self._auth_header, self.rid, request)
+        response = self._client._catalog_client.update_dataset_metadata(self._client._auth_header, self.rid, request)
 
-        dataset = self.__class__._from_conjure(self._auth_header, self._catalog_client, response)
+        dataset = self.__class__._from_conjure(self._client, response)
         update_dataclass(self, dataset, fields=self.__dataclass_fields__)
         return self
 
+    def add_to_dataset_from_io(
+        self,
+        dataset: BinaryIO,
+        timestamp_column_name: str,
+        timestamp_column_type: TimestampColumnType,
+        file_extension: _AllowedFileExtensions = ".csv",
+    ) -> None:
+        """Append to a dataset from a file-like object."""
+
+        if timestamp_column_type.startswith("relative"):
+            raise ValueError("multifile datasets with relative timestamps are not yet supported by the client library")
```

Comment:
I think we should check that the dataset is in the success state _before_ we start uploading the file - otherwise will have to re-upload the file on failure as we don't cache it anywhere

Reply:
done! good catch, removed this while testing.